Modules/_sqlite/cursor.c

```diff
@@ -750,47 +744,37 @@ pysqlite_cursor_executescript_impl(pysqlite_Cursor *self,
         return NULL;
     }
 
-    /* commit first */
-    result = _PyObject_CallMethodIdNoArgs((PyObject *)self->connection, &PyId_commit);
-    if (!result) {
-        goto error;
-    }
-    Py_DECREF(result);
-
-    pysqlite_state *state = self->connection->state;
-    while (1) {
-        const char *tail;
+    // Commit if needed
+    sqlite3 *db = self->connection->db;
+    if (!sqlite3_get_autocommit(db)) {
```

Comment:
I'm finding trouble trying to verify that this PR keeps behaviour identically. How is this condition checked in the old code?

Reply:
You'll find that in `pysqlite_connection_commit_impl()`:
https://github.com/python/cpython/blob/86d8b465231473f850cc5e906013ba8581ddb503/Modules/_sqlite/connection.c#L396-L436

Reply:
Ah, this is the piece I was missing